utils.groovy

```diff
@@ -516,26 +516,32 @@ def build_artifacts(pipecfg, stream, basearch) {
 
 def get_registry_repos(pipecfg, stream, version) {
     def registry_repos = pipecfg.registry_repos ?: [:]
-    // merge top-level registry_repos with stream-specific bits
-    registry_repos += pipecfg.streams[stream].additional_registry_repos ?: [:]
+    // get stream-specific bits additional repos
+    def additional_repos = pipecfg.streams[stream].additional_registry_repos ?: [:]
+    for (repo in ['oscontainer', 'legacy_oscontainer', 'extensions', 'kubevirt']) {
+        //merge top-level registry_repos with addtional repos
+        registry_repos[repo] += additional_repos[repo] ?: []
```

Comment:
We need to handle the case here where `repo` is not defined on `registry_repos` so it's not a list yet we can append to.